src/mgmt/srv_pool.c

```diff
@@ -378,15 +378,25 @@ ds_mgmt_pool_query(uuid_t pool_uuid, d_rank_list_t *svc_ranks, d_rank_list_t **r
 		   daos_pool_info_t *pool_info, uint32_t *pool_layout_ver,
 		   uint32_t *upgrade_layout_ver)
 {
+	uint64_t deadline;
+
 	if (pool_info == NULL) {
 		D_ERROR("pool_info was NULL\n");
 		return -DER_INVAL;
 	}
 
 	D_DEBUG(DB_MGMT, "Querying pool "DF_UUID"\n", DP_UUID(pool_uuid));
 
-	return ds_pool_svc_query(pool_uuid, svc_ranks, ranks, pool_info,
-				 pool_layout_ver, upgrade_layout_ver);
+	/*
+	 * Use a fixed timeout that matches what the control plane uses for the
+	 * moment.
+	 *
+	 * TODO: Pass the deadline from dmg (or daos_server).
```

Comment:
I can take a look at this if you want @liw .

Reply:
@tanabarr, thank you. That help is very welcome. Personally, I think when we start to work on such a ticket it would be helpful to review the deadline management and retry logic in dmg/daos_server (which I'm not familiar with), so that we could ensure dmg, daos_server, and daos_engine work well together.